mycore-base/src/main/java/org/mycore/datamodel/ifs2/MCRIFS2Commands.java

```diff
@@ -0,0 +1,220 @@
+        stores.map(MCRIFS2Commands::toString)
+            .forEach(LOGGER::info);
+    }
+
+    private static void initStores() {
+        initMetadataStores();
+        initFileStores();
+    }
+
+    private static void initFileStores() {
+        //FileSystems.getFileSystem(schemeURI) will not work in WebCLI
+        //Workaround is to go via MCRPath
+        final FileSystem defaultFileSystem = MCRPath.getRootPath("ignored").getFileSystem();
+        defaultFileSystem.getFileStores()
+            .forEach(FileStore::name);
+    }
+
+    private static void initMetadataStores() {
+        final MCRXMLMetadataManager metadataManager = MCRXMLMetadataManager.instance();
+        metadataManager.getObjectBaseIds().forEach(id -> {
+            final String[] parts = id.split("_");
+            metadataManager.getHighestStoredID(parts[0], parts[1]);
+        });
+    }
+
+    private static String toString(MCRStore store) {
+        final Map<String, String> configMap = MCRConfiguration2.getSubPropertiesMap(
+            "MCR.IFS2.Store." + store.getID() + ".");
+        final String config = configMap.entrySet().stream()
+            .sorted(Map.Entry.comparingByKey())
+            .map(e -> ", " + e.getKey() + "=" + e.getValue())
+            .collect(Collectors.joining());
+        return store.getClass().getSimpleName() + "{" +
+            "id=" + store.getID() +
+            ", baseDirectory=" + store.getBaseDirectory() +
+            ", lastID=" + store.getHighestStoredID() +
+            config +
+            '}';
+    }
+
+    @MCRCommand(syntax = "generate ifs2 md5sum files in directory {0}",
+        help = "writes md5sum files for every ifs2 file store in directory {0}")
+    public static List<String> writeMD5SumsToDirectory(String directory) throws NotDirectoryException {
+        Path targetPath = Paths.get(directory);
+        if (!Files.isDirectory(targetPath)) {
+            throw new NotDirectoryException(targetPath.toString());
+        }
+        initFileStores();
+        return MCRStoreCenter.instance().getCurrentStores(MCRFileStore.class)
+            .sorted(Comparator.comparing(MCRStore::getID))
+            .map(s -> "generate md5sum file " + targetPath.resolve(s.getID() + ".md5") + " for ifs2 file store "
+                + s.getID())
+            .collect(Collectors.toList());
+    }
+
+    @MCRCommand(syntax = "generate md5sum file {0} for ifs2 file store {1}",
+        help = "writes md5sum file {0} for every file in MCRFileStore with ID {1}")
+    public static void writeMD5SumFile(String targetFile, String ifsStoreId) throws IOException {
+        initFileStores();
+        final MCRStore store = MCRStoreCenter.instance().getStore(ifsStoreId);
+        if (!(store instanceof MCRFileStore)) {
+            LOGGER.error("Store " + ifsStoreId + " is not found or is not a file store.");
+            return;
+        }
+        Path targetPath = Paths.get(targetFile);
```

Comment:
What happens if the targetFile does not exist?

Reply:
It throws a NotDirectoryException, see next code line